layers/+lang/crystal/funcs.el

```diff
@@ -47,5 +47,9 @@
                         :variables company-tooltip-align-annotations t))))
 
 (defun spacemacs//crystal-setup-backend ()
-  "Conditionally setup crystal backend."
-  (when (eq crystal-backend 'lsp) (lsp)))
+  "Conditionally setup crystal backend and bindings."
+  (if (eq crystal-backend 'lsp)
+      (lsp)
+    (spacemacs/declare-prefix-for-mode 'crystal-mode "mg" "goto")
+    (spacemacs/declare-prefix-for-mode 'crystal-mode "mt" "test")
+    (spacemacs/declare-prefix-for-mode 'crystal-mode "ma" "action")))
```

Comment:
I know you just moved the prefix definition so this is not really related to this PR but I wonder why these prefixes are removed with the backend is `lsp`. 
Feels weird to me, `lsp` does not handle tests and it surely does not declare tests key bindings.

Reply:
This looks wrong, except of `test` these should not be bound when LSP is used. Also the prefixes should be set one time not everytime the mode is activated, this could otherwise degrade which-key's performance.